sys/arm64/vmm/vmm.c

```diff
@@ -842,8 +842,12 @@ vm_mmap_getnext(struct vm *vm, vm_paddr_t *gpa, int *segid,
 			*segoff = mmnext->segoff;
 		if (len)
 			*len = mmnext->len;
-		if (prot)
+		if (prot) {
 			*prot = mmnext->prot;
+#if __has_feature(capabilities)
```

Comment:
I don't think this needs to be conditional? Though stripping it off here feels a little sad, ideally we'd not have added it in the first place. But I don't understand the flow, and I know we have VM_OBJECT_MASK_CAP_PROT.

Reply:
It's conditional on `has_feature(capabilities)` because vm_mmap_memseg() implicitly adds VM_PROT_CAP when mapping memory into the guest address space, and it does so only when `has_feature(capabilities)` is true. I'll add a comment to make this clearer.

`VM_OBJECT_MASK_CAP_PROT` masks off `VM_PROT_CAP` if the VM object can't contain capabilities, but here it can (the object is backing guest memory), so that macro doesn't do what I want.

Reply:
My points were that (a) we unconditionally mask off VM_PROT_CAP in other places (b) VM_OBJECT_MASK_CAP_PROT's existence and use illustrates that the pattern of adding then later masking is already done despite it being sad

Reply:
I went ahead and removed the ifdef.